apps/aecore/lib/aecore/tx/data_tx.ex

```diff
@@ -331,11 +354,21 @@ defmodule Aecore.Tx.DataTx do
   end
 
   defp encode(tag, version, %DataTx{type: SpendTx} = tx) do
+    senders =
```

Comment:
This repeats for every transaction. Please separate into function.

_Only one comment for now. Full review hopefully tomorrow._